### ISSUE TITLE
Decorator not setup again after unrender/render cycle

### ISSUE THREAD

**Issue by sabob** [ractivejs/ractive#1904]
I'm using the Ractive sortable decorator and noticed that after I re-render views (unrender followed by render) the decorator is not instantiated again.

The teardown method of the decorator is invoked for the unrender call though.

Here is a fiddle:
Fiddle: http://fiddle.jshell.net/sabob/9g4tg97n/

The sortable decorator doesn't work with edge so I've added an updated copy in the fiddle itself. The actual demo is at the bottom of the JS section under the comment "Demo starts here".

The sortable list works when initially rendered, but after clicking the "Rerender" button the sorting is broken. The "Rerender" button performs a unrender/render of the Ractive.

**Comment by Madgvox** [comment]
This is unrelated to the bug, but from the console output, it looks like the entire decorator is being torn down and rebuilt every time the list is reordered, even when dragging. That... doesn't seem efficient.

**Comment by martypdx** [comment]
> it looks like the entire decorator is being torn down and rebuilt every time the list is reordered

The DOM nodes are being torndown and new one create on splice's, so decorator would be recreated.

**Comment by martypdx** [comment]
If you change it to a merge, then it doesn't teardown the nodes, it just shuffles (see http://fiddle.jshell.net/9g4tg97n/6/):

```js	
    array = ractive.get( sourceArray ).slice();
    source = array.splice( sourceIndex, 1 )[0];
    sourceIndex = targetIndex;
    array.splice( sourceIndex, 0, source );
    ractive.merge( sourceArray, array);
```

**Comment by Madgvox** [comment]
The issue in #1906 was fixed in 0.8-edge, and while the decorator itself is broken, it's not failing to be re-instantiated on render.
